File: app/src/main/java/com/gandhasiri/app/TrackerFragment.java
package com.gandhasiri.app;

import android.os.Bundle;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.fragment.app.Fragment;
import androidx.lifecycle.ViewModelProvider;
import androidx.navigation.Navigation;
import com.gandhasiri.app.databinding.FragmentTrackerBinding;

public class TrackerFragment extends Fragment {
    private FragmentTrackerBinding binding;
    private TreeViewModel treeViewModel;
    private TreeAdapter adapter;

    @Nullable
    @Override
    public View onCreateView(@NonNull LayoutInflater inflater, @Nullable ViewGroup container, @Nullable Bundle savedInstanceState) {
        binding = FragmentTrackerBinding.inflate(inflater, container, false);
        return binding.getRoot();
    }

    @Override
    public void onViewCreated(@NonNull View view, @Nullable Bundle savedInstanceState) {
        super.onViewCreated(view, savedInstanceState);
        
        adapter = new TreeAdapter();
        binding.rvTrees.setAdapter(adapter);

        adapter.setOnTreeClickListener(tree -> {
            Bundle bundle = new Bundle();
            bundle.putInt("treeId", tree.getId());
            bundle.putString("treeUid", tree.getTreeId());
            bundle.putString("photoPath", tree.getPhotoPath());
            bundle.putFloat("girth", (float) tree.getGirth());
            bundle.putFloat("lat", (float) tree.getLatitude());
            bundle.putFloat("lng", (float) tree.getLongitude());
            
            Navigation.findNavController(view).navigate(R.id.navigation_tree_detail, bundle);
        });

        treeViewModel = new ViewModelProvider(this).get(TreeViewModel.class);
        treeViewModel.getAllTrees().observe(getViewLifecycleOwner(), trees -> {
            if (trees != null) {
                adapter.setTrees(trees);
            }
        });
    }

    @Override
    public void onDestroyView() {
        super.onDestroyView();
        binding = null;
    }
}

File: app/src/main/java/com/gandhasiri/app/TrackerFragment.kt
package com.gandhasiri.app

import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.fragment.app.Fragment
import androidx.lifecycle.ViewModelProvider
import androidx.navigation.fragment.findNavController
import com.gandhasiri.app.databinding.FragmentTrackerBinding

class TrackerFragment : Fragment() {
    private var _binding: FragmentTrackerBinding? = null
    private val binding get() = _binding!!
    private lateinit var treeViewModel: TreeViewModel
    private lateinit var adapter: TreeAdapter

    override fun onCreateView(
        inflater: LayoutInflater,
        container: ViewGroup?,
        savedInstanceState: Bundle?
    ): View {
        _binding = FragmentTrackerBinding.inflate(inflater, container, false)
        return binding.root
    }

    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
        super.onViewCreated(view, savedInstanceState)
        
        adapter = TreeAdapter()
        binding.rvTrees.adapter = adapter

        adapter.setOnTreeClickListener { tree ->
            val bundle = Bundle().apply {
                putInt("treeId", tree.id)
                putString("treeUid", tree.treeId)
                putString("photoPath", tree.photoPath)
                putFloat("girth", tree.girth.toFloat())
                putFloat("lat", tree.latitude.toFloat())
                putFloat("lng", tree.longitude.toFloat())
            }
            findNavController().navigate(R.id.navigation_tree_detail, bundle)
        }

        treeViewModel = ViewModelProvider(this).get(TreeViewModel::class.java)
        treeViewModel.allTrees.observe(viewLifecycleOwner) { trees ->
            trees?.let {
                adapter.setTrees(it)
            }
        }
    }

    override fun onDestroyView() {
        super.onDestroyView()
        _binding = null
    }
}
